modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/HeadersWithValidationResult.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http.netty4;
+
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import io.netty.handler.codec.http.DefaultHttpRequest;
+import io.netty.handler.codec.http.HttpHeaders;
+
+import org.apache.lucene.util.SetOnce;
+
+public class HeadersWithValidationResult extends DefaultHttpHeaders {
+
+    private final SetOnce<HeadersValidationResult> validationResult = new SetOnce<>();
+
+    public static DefaultHttpRequest wrapRequestWithValidatableHeaders(DefaultHttpRequest originalRequest, boolean validateHeaders) {
+        HeadersWithValidationResult httpHeaders = new HeadersWithValidationResult(originalRequest.headers(), validateHeaders);
+        return new DefaultHttpRequest(originalRequest.protocolVersion(), originalRequest.method(), originalRequest.uri(), httpHeaders);
+    }
+
+    public HeadersWithValidationResult(HttpHeaders httpHeaders, boolean validateHeaders) {
+        super(validateHeaders);
+        add(httpHeaders);
```

Comment:
The outcome here is that the `HeadersWithValidationResult` will contain a `DefaultHeadersImpl` for the `DefaultHttpMessage#headers` field, whereas the "wrapped" instance from the `httpHeaders` argument, if it is a `DefaultHttpHeaders` instance as well, might contain a different implementation such as `CombinedHttpHeaders`. BUT looking at the current codebase, that's only a theoretical possibility, in practice it should always also be a `DefaultHeadersImpl`.

Even if, following some future netty lib version update, the implementatin type changes when we wrap, it still won't 

But I wanted to call out this "brittleness" source: It's possible that, following some future netty lib version upgrade, the change of type due to the wrapping here might cause problems, i.e. the original `Headers` implementation might have a slightly different behavior to the one of the wrapper, or downstream code might use `instance of` checks, which break when we wrap types.